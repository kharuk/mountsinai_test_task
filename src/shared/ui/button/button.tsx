import * as React from 'react';
export interface Props {
  children?: React.ReactNode;
  onClick?: (e: any) => void;
  styles?: {};
}

export const Button: React.FC<Props> = (props) => (
  <button onClick={props.onClick} style={props.styles} type="button">
    {props.children}
  </button>
);
