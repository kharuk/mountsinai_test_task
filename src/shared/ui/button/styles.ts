import styled from 'styled-components';
import { colors } from 'shared/variables/colors';

export const DefaultButton = styled.button`
  box-sizing: border-box;
  display: flex;
  align-items: center;
  justify-content: center;
  vertical-align: middle;
  padding: 0.5rem 1rem;
  pointer-events: 'auto';
  text-decoration: none;
  background: transparent;
  outline: none;
  transition: 0.5s;

  &:focus {
    outline: none;
  }

  &:hover {
    transition: 0.5s;
    cursor: pointer;
  }

  &:disabled {
    cursor: default;
    pointer-events: none;
  }

  &:active {
    background-color: ${colors.primary};
    opacity: 0.5;
  }
`;

export const RoundButton = styled(DefaultButton)`
  border-radius: 50%;
  background: ${colors.primary};
  opacity: 0.4;
  width: 60px;
  height: 60px;
  transition: transform 0.3s ease-out;

  &:hover {
    background-color: ${colors.primary};
    opacity: 0.3;
    transform: scale(1.1);
  }

  &:disabled {
    background-color: ${colors.disabled};
  }
  &:active {
    background-color: ${colors.primary};
    opacity: 0.5;
  }

  svg {
    color: ${colors.white};
  }
`;
