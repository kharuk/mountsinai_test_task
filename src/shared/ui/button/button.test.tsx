import React from 'react';
import { shallow } from 'enzyme';
/* import 'jest-styled-components'; */
import { Button } from './button';

describe('Button component', () => {
  it('Renders button', () => {
    const button = shallow(<Button>button</Button>);
    expect(button).toMatchSnapshot();
  });
});
