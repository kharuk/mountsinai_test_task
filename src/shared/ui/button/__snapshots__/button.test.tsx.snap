// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component Renders button 1`] = `ShallowWrapper {}`;
